dump_dbus_util.cpp

```diff
@@ -110,4 +97,99 @@ ManagedObjectType getDumpEntries(sdbusplus::bus::bus& bus)
     }
     return objects;
 }
+
+bool isSystemHMCManaged(sdbusplus::bus::bus& bus)
+{
+    using BiosBaseTableItem = std::pair<
+        std::string,
+        std::tuple<std::string, bool, std::string, std::string, std::string,
+                   std::variant<int64_t, std::string>,
+                   std::variant<int64_t, std::string>,
+                   std::vector<std::tuple<
+                       std::string, std::variant<int64_t, std::string>>>>>;
+    using BiosBaseTable = std::vector<BiosBaseTableItem>;
+    try
+    {
+        std::string hmcManaged{};
+        auto retVal = readDBusProperty<std::variant<BiosBaseTable>>(
+            bus, "xyz.openbmc_project.BIOSConfigManager",
+            "/xyz/openbmc_project/bios_config/manager",
+            "xyz.openbmc_project.BIOSConfig.Manager", "BaseBIOSTable");
+        const auto baseBiosTable = std::get_if<BiosBaseTable>(&retVal);
+        if (baseBiosTable == nullptr)
+        {
+            std::string err =
+                "Failed to read BIOSconfig property BaseBIOSTable";
+            log<level::ERR>(err.c_str());
+            throw std::runtime_error(err);
+        }
+        for (const auto& item : *baseBiosTable)
+        {
+            std::string attributeName = std::get<0>(item);
+            auto attrValue = std::get<5>(std::get<1>(item));
+            auto val = std::get_if<std::string>(&attrValue);
+            if (val != nullptr && attributeName == "pvm_hmc_managed")
+            {
+                hmcManaged = *val;
+                break;
+            }
+        }
+        if (hmcManaged.empty())
+        {
+            log<level::ERR>("Failed to read pvm_hmc_managed property value");
+            return false;
+        }
+        if (hmcManaged == "Enabled")
+        {
+            return true;
+        }
+    }
+    catch (const std::exception& ex)
+    {
+        log<level::ERR>(
+            fmt::format("Failed to read pvm_hmc_managed property ({})",
+                        ex.what())
+                .c_str());
+        return false;
+    }
+
+    return false;
+}
+
+bool isHostRunning(sdbusplus::bus::bus& bus)
+{
+    try
+    {
+        constexpr auto hostStateObjPath = "/xyz/openbmc_project/state/host0";
+        auto retVal = readDBusProperty<DBusProgressValue_t>(
+            bus, "xyz.openbmc_project.State.Host", hostStateObjPath,
+            "xyz.openbmc_project.State.Boot.Progress", "BootProgress");
+        const std::string* progPtr = std::get_if<std::string>(&retVal);
+        if (progPtr == nullptr)
+        {
+            std::string err = fmt::format(
```

Comment:
You could directly pass the error message in the `log<>`

Reply:
i am adding hoststate object here

Reply:
I am not following, you mean formatting error message with arguments? If so, you can do it like other places (added the exception)